client/src/main/java/org/asynchttpclient/netty/NettyResponse.java

```diff
@@ -179,6 +179,7 @@ public ByteBuffer getResponseBodyAsByteBuffer() {
         for (HttpResponseBodyPart part : bodyParts)
             target.put(part.getBodyPartBytes());
 
+        target.rewind();
```

Comment:
Well, `flip` is the idiomatic way for switching from write to read and vice versa, which is exactly what we need here.